### DISCUSSION TITLE
Occured "Uncaught ReferenceError" after reloading component on x-bind:class only

### DISCUSSION THREAD

**Discussion by mayanksdudakiya** [alpinejs/alpine#3299] [7. Bugs (archived)]
@calebporzio With reference of https://github.com/alpinejs/alpine/issues/825 of Alpine issue. I am still facing this issue with the Alpine version `3.10.3`. So once the page successfully loaded first time then there was no any console error but when I tried to apply any filters and filter will reload the component at that time it shows the `Uncaught ReferenceError: is_theme_tab_opened_1 is not defined` error. Please have a look at the below screenshots for better understanding. 

Issue is happening with `x-bind:class` only

### Dependencies
![image](https://user-images.githubusercontent.com/34036151/192465333-b8424fa1-e525-4859-b7af-f5b9e45b9fd1.png)

### Code
`x-data` has already been defined but it still shows reference error in the browser console.

![image](https://user-images.githubusercontent.com/34036151/192465500-3e63a238-3333-4d07-9b86-20d179858983.png)

### Browser console error
![image](https://user-images.githubusercontent.com/34036151/192466045-25cdd8ad-0775-4a87-86b6-65316df9a55d.png)

### Inspected element with `x-data` already defined in <li>
![image](https://user-images.githubusercontent.com/34036151/192477328-372cd3d7-fd70-49d1-8c27-45e58096b428.png)

_Originally posted by @mayanksdudakiya in https://github.com/alpinejs/alpine/discussions/2114#discussioncomment-3740850_

**Comment by joshhanley** [comment]
This issue has been converted to a discussion as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/alpinejs/alpine/discussions/1423
